prow/cmd/checkconfig/main.go

```diff
@@ -935,8 +935,18 @@ func validateTriggers(cfg *config.Config, pcfg *plugins.Configuration) error {
 	configured := newOrgRepoConfig(map[string]sets.String{}, configuredRepos)
 	enabled := enabledOrgReposForPlugin(pcfg, trigger.PluginName, false)
 
-	if missing := configured.difference(enabled).items(); len(missing) > 0 {
-		return fmt.Errorf("the following repos have jobs configured but do not have the %s plugin enabled: %s", trigger.PluginName, strings.Join(missing, ", "))
+	missing := configured.difference(enabled).items()
+	var missNotGerrit []string
+	if len(missing) > 0 {
+		for _, repo := range missing {
+			if !strings.Contains(repo, "googlesource.com") {
```

Comment:
Is there a more generic way to detect a `gerrit` job? Do we furthermore need to then validate that whatever is making `gerrit` work (in stead of `trigger`) is turned on?

Reply:
Will defer to @cjwagner on this

Reply:
> Is there a more generic way to detect a `gerrit` job? 

I'm not sure. It would be nice if there is, but I don't know of one. @krzyzacy do you have any suggestions?

> Do we furthermore need to then validate that whatever is making `gerrit` work (in stead of `trigger`) is turned on?

Ideally we would, but the configuration for which repos to scrape is passed to the gerrit controller via command line arguments instead of config.yaml. I think if we care about validating that we should first switch to specifying the gerrit repos to scrape in config.yaml. I don't think that is a blocker for addressing this bug though.

Reply:
there's also https://github.com/GoogleCloudPlatform/oss-test-infra/issues/103

I would suggest we add a dedicated prowjob field/label to specify the source of the prowjob? (github or gerrit?)

Reply:
gerrit reporter checks for https://github.com/kubernetes/test-infra/blob/master/prow/gerrit/reporter/reporter.go#L110-L116, which might make more sense.

Reply:
yeah let's key off of those annotations if we expect them to be set for all Gerrit jobs. What should we validate instead of `trigger`? How do gerrit jobs get started?

Reply:
there's a gerrit controller. (trigger equivalent)